toolkit/tools/pkg/imagecustomizerlib/runscripts_test.go

```diff
@@ -66,3 +66,37 @@ resolv.conf exists
 
 	verifyFileContentsSame(t, aOrigFilePath, aNewFilePath)
 }
+
+func TestCustomizeImageRunScriptsIptables(t *testing.T) {
+	var err error
+
+	baseImage := checkSkipForCustomizeImage(t, baseImageTypeCoreEfi, baseImageVersionDefault)
+
+	testTmpDir := filepath.Join(tmpDir, "TestCustomizeImageRunScriptsIptables")
+	buildDir := filepath.Join(testTmpDir, "build")
+	configFile := filepath.Join(testDir, "runscripts-iptables.yaml")
+	outImageFilePath := filepath.Join(testTmpDir, "image.raw")
+
+	// Customize image.
+	err = CustomizeImageWithConfigFile(buildDir, configFile, baseImage, nil, outImageFilePath, "raw", "",
+		"" /*outputPXEArtifactsDir*/, false /*useBaseImageRpmRepos*/, false /*enableShrinkFilesystems*/)
+	assert.ErrorContains(t, err, "failed to customize raw image")
+	assert.ErrorContains(t, err, "script (postCustomization[0]) failed")
+}
+
+func TestCustomizeImageRunScriptsModprobe(t *testing.T) {
+	var err error
+
+	baseImage := checkSkipForCustomizeImage(t, baseImageTypeCoreEfi, baseImageVersionDefault)
+
+	testTmpDir := filepath.Join(tmpDir, "TestCustomizeImageRunScriptsModprobe")
+	buildDir := filepath.Join(testTmpDir, "build")
+	configFile := filepath.Join(testDir, "runscripts-modprobe.yaml")
+	outImageFilePath := filepath.Join(testTmpDir, "image.raw")
+
+	// Customize image.
+	err = CustomizeImageWithConfigFile(buildDir, configFile, baseImage, nil, outImageFilePath, "raw", "",
+		"" /*outputPXEArtifactsDir*/, false /*useBaseImageRpmRepos*/, false /*enableShrinkFilesystems*/)
+	assert.ErrorContains(t, err, "failed to customize raw image")
+	assert.ErrorContains(t, err, "script (postCustomization[0]) failed")
```

Comment:
Is the propagated error more descriptive? in other words, will the user understand what they are doing incorrectly?

Reply:
We also report the last line of stderr in the error. So, it depends on the command that fails and how good its error messages are.

I didn't include the stderr messages here in case they are inconsistent between platforms/versions.